go-controller/pkg/ovn/master.go

```diff
@@ -1248,9 +1259,14 @@ func (oc *Controller) syncNodes(nodes []interface{}) {
 		chassisMap[chassis.Hostname] = chassis.Name
 	}
 
-	//delete existing nodes from the chassis map.
-	for nodeName := range foundNodes {
-		delete(chassisMap, nodeName)
+	// If chassis annotation exists and is the same as the chassis in the SBDB, delete it from the chassis map.
+	// If chassis has changed or the node has been deleted when ovnkube-master is down, it will stay in the chassis
+	// map and will then be deleted in the SBDB.
+	for nodeName, chassisAnnotation := range chassisAnnotationMap {
+		chassis, ok := chassisMap[nodeName]
+		if ok && (chassisAnnotation == "" || chassis == chassisAnnotation) {
+			delete(chassisMap, nodeName)
+		}
 	}
```

Comment:
Why do we need these L1265-1269... don't we do the samething in addNodes()? 

After syncNodes() is called, we call addNodes() for each of the node...